runtime/tracer/track_memory_map.c

```diff
@@ -817,7 +836,7 @@ bool track_memory_map(pid_t pid, int *exit_status_out, enum trace_mode mode) {
     case WAIT_SIGNALED: {
       fprintf(stderr, "process received fatal signal (syscall entry)\n");
       enum control_flow cf = handle_process_exit(&maps, waited_pid);
-      return false;
+      return true;
```

Comment:
Making this return `true` without setting `*exit_status_out` violates the doc comment for `track_memory_map`. This needs changes in `wait_for_next_trap`, which currently only writes through its copy of the `exit_status_out` pointer when returning `WAIT_EXITED`

Reply:
I changed things to always return the `wait` status, which can then be inspected more thoroughly with all of the `W*` macros.

Reply:
This should still not `return true;`; doing so unconditionally exits even if other processes are still around and being traced. I actually don't understand why this case is being changed at all, given `WAIT_PASSTHROUGH_SIGNAL` is being introduced here as distinct from `WAIT_SIGNALED`. The latter used to mean "an *unexpected* signal was received", so it was handled by exiting from the entire function with failure. What situation is now being handled by this block?